curriculum/challenges/english/25-front-end-development/quiz-javascript-functions/66edcc779993c0da6906dbb9.md

```diff
@@ -17,439 +17,482 @@ Answer all of the questions below correctly to pass the quiz.
+`function = myFunction();`
 
 #### --answer--
 
-Placeholder answer
+`function myFunction() {}`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is a valid function expression?
 
 #### --distractors--
 
-Placeholder distractor 1
+```js
+function getSum(x, y) { 
+    return x + y; 
+};
+```
 
 ---
 
-Placeholder distractor 2
+```js
+getSum: function(x, y) { 
+    return x + y; 
+};
+```
 
 ---
 
-Placeholder distractor 3
+```js
+function = getSum(x, y) { 
+    return x + y; 
+};
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+getSum = function(x, y) { 
+    return x + y; 
+};
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Where can function expressions be used?
 
 #### --distractors--
 
-Placeholder distractor 1
+Only within other functions
 
 ---
 
-Placeholder distractor 2
+Only as part of an object literal
 
 ---
 
-Placeholder distractor 3
+Only in the global scope
 
 #### --answer--
 
-Placeholder answer
+Anywhere a variable can be used
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the difference between a function expression and a function declaration?
 
 #### --distractors--
 
-Placeholder distractor 1
+Function expressions can be hoisted, while function declarations cannot.
 
 ---
 
-Placeholder distractor 2
+Both function expressions and function declarations can be hoisted.
 
 ---
 
-Placeholder distractor 3
+Neither function expressions nor function declarations can be hoisted.
 
 #### --answer--
 
-Placeholder answer
+Function declarations can be hoisted, while function expressions cannot.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does it mean for functions to be first-class citizens in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+They can be assigned to variables.
 
 ---
 
-Placeholder distractor 2
+They can be passed as arguments to other functions.
 
 ---
 
-Placeholder distractor 3
+They can be returned from functions.
 
 #### --answer--
 
-Placeholder answer
+All of the answers are correct.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is a common use case for first-class functions in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+Implementing object-oriented programming
 
 ---
 
-Placeholder distractor 2
+Defining global variables
 
 ---
 
-Placeholder distractor 3
+Creating custom data types
 
 #### --answer--
 
-Placeholder answer
+Creating higher-order functions
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following is an example of a first-class function?
 
 #### --distractors--
 
-Placeholder distractor 1
+```js
+function greet(name) { 
+    console.log("Hello, " + name); 
+}
+```
 
 ---
 
-Placeholder distractor 2
+```js
+var greet = function(name) { 
+    console.log("Hello, " + name); 
+};
+```
 
 ---
 
-Placeholder distractor 3
+```js
+greet: function(name) { 
+    console.log("Hello, " + name); 
+}
+```
```

Comment:
On it own this will get `Uncaught SyntaxError: Function statements require a function name`. Maybe replace it with arrow function?

Reply:
fixed